pkg/daemon/update.go

```diff
@@ -169,14 +169,18 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr err
 		}
 	}()
 
-	oldConfigName := oldConfig.GetName()
-	newConfigName := newConfig.GetName()
-	glog.Infof("Checking Reconcilable for config %v to %v", oldConfigName, newConfigName)
-	// We skip out of Reconcilable if there is no Kind and we are in runOnce mode. The
-	// reason is that there is a good chance a previous state is not available to match against.
-	if oldConfig.Kind == "" && dn.onceFrom != "" {
-		glog.Info("Missing kind in old config. Assuming no prior state.")
+	oldConfigUnset := oldConfig == nil
+	if oldConfigUnset {
+		// Rather than change the rest of the code to deal
+		// with a nil oldConfig, just pass an empty one in
+		// most places.
+		emptyMC := mcfgv1.MachineConfig{}
+		oldConfig = &emptyMC
```

Comment:
note for next time (don't have to fix in this pr): can remove `emptyMC` declaration and just do `oldConfig = &mcfgv1.MachineConfig{}` directly